ComfortZoneApp/app/src/main/java/com/example/comfortzone/utils/ComfortCalcUtil.java

```diff
@@ -59,19 +62,27 @@ private static int getSum(LevelsTracker tracker) {
 
     public static void calculateAverages(ParseUser currentUser) {
         ArrayList<LevelsTracker> trackerArrayList = (ArrayList<LevelsTracker>) currentUser.get(KEY_LEVEL_TRACKERS);
-        trackerArrayList.forEach(tracker -> calculateSingularAverage(tracker));
+        List<Task<Void>> savingAverages = trackerArrayList.stream().filter(tracker -> tracker.getCount() > 0).map(tracker -> calculateSingularAverage(tracker)).collect(Collectors.toList());
+        Task.whenAll(savingAverages).onSuccess(new Continuation<Void, Object>() {
```

Comment:
would there also be a case when saving averages fails and do we need to handle that?